sources/osgShadow/ShadowMap.js

```diff
@@ -0,0 +1,691 @@
+            zFar = this._farCaster;
+            zNear = this._nearCaster;
+            zFix = true;
+
+            if ( camera.getComputeNearFar() === false || zNear === undefined || zFar === undefined ) {
+                // didn't get near/far auto computed
+                // first frame, off, etc.
+                frustumBound = this.getShadowedScene().getBound();
+                center = frustumBound.center();
+                radius = frustumBound.radius();
+
+                centerDistance = Vec3.distance( worldLightPos, center );
+                zNear = centerDistance - radius;
+                zFar = centerDistance + radius;
+                zFix = false;
+            }
+
+            // Empty or Bad Frustums
+            // No objects, handle it gracefully
+            var epsilon = 1e-6;
+            if ( zFar < zNear - epsilon ) {
+
+                // TODO: clear shadow texture and return wihtout any further ops
+                // with a early out
+                Notify.log( 'empty shadowMap' );
+                // for now just prevent NaN errors
+
+                zFar = 1;
+                zNear = 0.001;
+                zFix = true;
+            }
+
+            // Check div by zero or NaN cases
+            if ( zNear === Number.POSITIVE_INFINITY || zNear < epsilon ) {
+                zNear = epsilon;
+                zFix = true;
+            }
+            if ( zNear === Number.NEGATIVE_INFINITY || zFar < epsilon ) {
+                zFar = 1.0;
+                zFix = true;
+            }
+
+            var zNearRatio = 0.001;
+            if ( zNear < zFar * zNearRatio ) {
+                zNear = zFar * zNearRatio;
+                zFix = true;
+            }
+
+            if ( zFix ) {
+                radius = ( zFar - zNear ) * 0.5;
+
+
+                // move from camera to near
+                // first compute a ray along light dir of length near
+                Vec3.mult( worldLightDir, zNear, this._tmpVec );
+                // then
+                Vec3.add( worldLightPos, this._tmpVec, this._tmpVec );
+
+                // move from near to center
+                // first compute a ray along light dir of length radius
+                Vec3.mult( worldLightDir, radius, this._tmpVecBis );
+                Vec3.add( this._tmpVec, this._tmpVecBis, this._tmpVec );
+
+                center = this._tmpVec;
+                centerDistance = Vec3.distance( worldLightPos, center );
+
+                zNear = centerDistance - radius;
+                zFar = centerDistance + radius;
+
+            }
+
+            Notify.assert( zNear > epsilon );
+            Notify.assert( zFar > zNear + epsilon );
+
+            var top, right;
+            var up = this._lightUp;
+
+            if ( Math.abs( Vec3.dot( up, worldLightDir ) ) >= 1.0 ) {
+                // another camera up
+                up = [ 1.0, 0.0, 0.0 ];
+            }
+
+            if ( light.getPosition()[ 3 ] !== 0.0 ) {
```

Comment:
Hemi doesn't have a light position layout like the direction ?
( on other word isDirection() on a hemi doesn't return true ? )